### PR TITLE
[performance_optim] define flash attention mask on NPU device directly

### PR DESCRIPTION
**Pull request by FightingZhen**
# What does this PR do?

When using Flash Attention2 on Ascend NPU, we have found that CPU memory keep increasing when calling func `npu_flash_attn_varlen_func` or `npu_flash_attn_func`. 

The root cause is that the attention mask generated by func `torch.ones()` is initially defined on the CPU side, occupying CPU memory before being transferred to the NPU device. As the func `npu_flash_attn_varlen_func` or `npu_flash_attn_func` is called repeatedly, the CPU memory consumption continues to accumulate, which is not optimal solution. Below is one example: https://github.com/huggingface/transformers/blob/12f65ee7520404b511c7fe716bc5d48d93d8297d/src/transformers/integrations/npu_flash_attention.py#L225

Therefore, this PR is committed for solving this problem by defining attention mask tensor with `torch.ones()` on NPU device directy.

Fixes # (issue)
Not releated.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by FightingZhen** [comment]
@MekkCyber @SunMarc please help me review it, thanks : )

**Review by MekkCyber — APPROVED** [review]
Sounds good ! thanks for catching that 🤗

**Review by SunMarc — APPROVED** [review]
SGTM !

**Comment by FightingZhen** [comment]
@MekkCyber @SunMarc I have rebased the code to `main` branch and repushed it, I am not clear why `build_pr_documentation` CI pending, and the CI error seems not related with my PR, and  can you please help me merge this PR? thanks : )

**Comment by FightingZhen** [comment]
@MekkCyber @SunMarc All CI passes and this PR seems to be ready for merge :)

**Comment by MekkCyber** [comment]
Merged 🎊 ! Thanks a lot
